docs/rfcs/016-asset-metadata.md

```diff
@@ -0,0 +1,95 @@
+  * If we opted for this table we could consider adding a new column to the [`ImageDeliveryChannel`](014-delivery-channels-database.md) table, as this would store a row per Asset/Delivery channel.
+* `AssetDeliveryMetadata` - similar to above - if we are storing checksum etc this isn't asset delivery (ie Orchestrator/Thumbs) specific.
+
+#### Schema
+
+The suggested schema for the table should be flexible.
+
+| Column        | Type        | Description                         |
+| ------------- | ----------- | ----------------------------------- |
+| AssetId       | text        | AssetId this is for                 |
+| MetadataType  | text        | Identifier for the type of metadata |
+| MetadataValue | jsonb       | JSON object of values for type      |
+| {audit-cols}  | timestamptz | Created/updated dates               |
+
+* `AssetId` - this maintains link back to asset. Storing Id only is fine, no need to store separate `customer` or `space` as lookup will only be by Id.
+* `MetadataType` is the 'key' used to look up relevant metadata - these values won't link to anything in the database but a known list of values will be maintained and used by the application code.
+* `MetadataValue` is a `jsonb` value storing relevant data as JSON. In most cases I think we would always want to read the entire object but it could be useful to have efficient querying afforded by `jsonb` (e.g. to get `"o"`pen thumbs only). This querying has support in [npgsql](https://www.npgsql.org/efcore/mapping/json.html).
+* `AssetId` and `MetadataType` would be composite key.
+
+#### Sample Data
+
+Example data for an asset could be:
+
+| AssetId | MetadataType | MetadataValue                                                     |
+| ------- | ------------ | ----------------------------------------------------------------- |
+| 1/2/foo | ThumbSizes   | `{"o": [[200,127],[100,64]],"a": [[1024,651], [400,254]]}`        |
+| 1/2/foo | AVTranscodes | `["1/2/foo/full/max/default.mp3","1/2/foo/full/max/default.avi"]` |
+| 1/2/foo | Checksum     | `{"sha256": "abc123123123"}`                                      |
+
+`ThumbSizes` is the only type we're interested in now but the other values are indicative of what we could store.
+
+#### Querying / EF Entities
+
+Objects could be included where required via filtered include statement to filter on `MetadataType`. 
+
+```cs
+var assetWithThumbs = dbContext.Images
+                .Include(i => i.AssetApplicationMetadata.Where(m => m.MetadataType == "ThumbSizes"))
+                .Single(i => i.Id == assetId);
+```
+
+## Outstanding Questions
+
+* For `Asset` object - would we use polymorphic deserialisation? How complex would that be? Would storing on object as a `string` be okay, deserialisation as required (rather than on way out of DB)
+* Should we store sizes in s3 as well as db?

```

Comment:
I think so, also for making thumb serving independent of DB schema, self-contained.